### PR TITLE
cutter: fix build; switch from qt5 to qt6

### PR DESCRIPTION
**Pull request by chayleaf**
## Description of changes

This depends on https://github.com/NixOS/nixpkgs/pull/265987

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lilyinstarlight** [comment]
Just curious, what issues were you getting that you needed to change the `BUILD_TARGET` and `INSTALL_TARGET` stuff upstream?

I really only needed the patch from https://github.com/NixOS/nixpkgs/pull/263951#issuecomment-1797306203 but also needed the upstream commit from https://github.com/NixOS/nixpkgs/pull/265987/files#diff-44382681a0796856d63801d3a010b4c02bdf755c310bd6df2b51b8aa6dee7424R28-R34 since there was a regression in shiboken 6.6.0 (which I'm also unsure how you managed to build it without that)

**Comment by chayleaf** [comment]
I have no idea, I just tried to build it and looked at what include files were missing

**Comment by lilyinstarlight** [comment]
@ofborg eval

Diff looks good to me. I'm requesting a fresh ofborg eval since the other PR got merged and will merge if it's all green

Thank you!

**Comment by chayleaf** [comment]
yes i'll just rebase this pr on master

**Comment by chayleaf** [comment]
huh, there's an API extractor issue. Could the 6.6.0 patch have broken it?

**Comment by lilyinstarlight** [comment]
Darwin needs https://github.com/rizinorg/cutter/issues/3188#issuecomment-1829657417 to work, just importing the other patch is not enough

I'm inclined to merge this anyway since I mean the build is broken now as-is and whenever upstream fixes it, we can pull it in and backport it too

File: pkgs/development/tools/analysis/rizin/cutter.nix
Comment (lilyinstarlight) [inline pkgs/development/tools/analysis/rizin/cutter.nix:40]:
```suggestion
    (fetchpatch {
      name = "cutter-simplify-python-binding-include-handling.patch";
      url = "https://github.com/rizinorg/cutter/compare/7256fbb00e92ab12a24d14a92364db482ed295cb..ca5949d9d7c907185cf3d062d9fa71c34c5960d4.diff";
      hash = "sha256-bqV2FTA8lMNpHBDXdenNx+1cLYa7MH47XKo1YatmLV4=";
    })
```

Use all three commits from upstream PR now

File: pkgs/development/tools/analysis/rizin/cutter.nix
Comment (lilyinstarlight) [inline pkgs/development/tools/analysis/rizin/cutter.nix:93]:
```suggestion
```

Darwin is fixed 🎉

File: pkgs/development/tools/analysis/rizin/cutter.nix
Comment (lilyinstarlight) [inline pkgs/development/tools/analysis/rizin/cutter.nix:99]:
```suggestion
    inherit (rizin.meta) platforms;
```

Eval fix

**Review by lilyinstarlight — COMMENTED** [review]
Upstream has updated rizinorg/cutter#3268 and it seems to work now for Darwin build. After these changes, this PR should be in a merge-ready state

Thank you again!

**Review by lilyinstarlight — APPROVED** [review]
Diff looks fabulous, I'll merge when ofborg is green

Edit: Ofborg darwin builds timed out due to Qt 6.6.1 rebuild on master, but I've manually verified myself that this builds on aarch64-darwin and x86_64-darwin

**Pull request merged**
